code/controllers/subsystems/chemistry.dm

```diff
@@ -7,10 +7,51 @@ var/datum/controller/subsystem/chemistry/SSchemistry
 
 	var/list/active_holders = list()
 	var/list/chemical_reactions
+	var/list/chemical_reactions_clean = list()
 	var/list/chemical_reagents
 
 	var/tmp/list/processing_holders = list()
 
+/datum/controller/subsystem/chemistry/proc/check_specific_heat(var/datum/reagent/R,var/unit_test = FALSE)
```

Comment:
This function is not sane. Returning a number or a boolean is not a smart way to set up an API. Have a fallback value (0 or 1) or crash the function at illogical moments.